src/pages/generalSettings/IntelligentAgentSettings.jsx

```diff
@@ -0,0 +1,169 @@
+import DividerTitle from '../../components/DividerTitle';
+import Link from '../../components/Link';
+
+export default function IntelligentAgentSettings({
+  intelligentAgentSettingsFields,
+  currentValues,
+  setCurrentValues,
+  siteSettings,
+  setAllValid,
+}) {
+  return intelligentAgentSchema.map(intelligentAgent => {
+    const currentPlatformFields = get(
+      intelligentAgent,
+      ['data', 'fields'],
+      [],
+    );
+    const currentPlatformEnablingField = get(intelligentAgent, [
+      'data',
+      'enablingField',
+    ]);
+
+    const isCurrentPlatformEnabled = get(
+      currentValues,
+      currentPlatformEnablingField,
+      false,
+    );
+
+    const noCredsMissing =
+      intelligentAgentSettingsFields.filter(
+        currentField => get(currentValues, currentField) === '',
+      ).length === 0;
+
+    const allValid =
+      !isCurrentPlatformEnabled ||
+      (isCurrentPlatformEnabled && noCredsMissing);
+    setAllValid(allValid);
+    const enablingPlatformField = currentPlatformFields.find(
+      currentField => {
+        const settingKey = get(currentField, 'label');
+        return settingKey === currentPlatformEnablingField;
+      },
+    );
+    const enabledSettingKey = get(enablingPlatformField, 'label');
+    const enabledSkipDescription = get(
+      enablingPlatformField,
+      'skipDescription',
+      false,
+    );
+
+    if (isCurrentPlatformEnabled) {
+      return (
+        <>
+          <SettingsTextInput
+            key={enabledSettingKey}
+            settingKey={enabledSettingKey}
+            customFieldCategories={[]}
+            currentValues={currentValues}
+            setCurrentValues={setCurrentValues}
+            siteSettings={siteSettings}
+            skipDescription={enabledSkipDescription}
+          />
+          <DividerTitle
+            titleId={get(intelligentAgent, 'dividerLabel')}
+            style={{ marginTop: 32 }}
+          />
+          <Text
+            style={{ marginTop: 20 }}
+            variant="subtitle1"
+            id={get(intelligentAgent, 'platformInformationLabel')}
+          />
+          <Text
+            style={{ marginTop: 4 }}
+            variant="body2"
+            id={get(
+              intelligentAgent,
+              'platformInformationDescription',
+            )}
+            values={{
+              documentationLink: (
+                <Link
+                  newTab
+                  external
+                  href={get(intelligentAgent, 'apiDocumentationUrl')}
+                >
+                  <FormattedMessage
+                    id={get(
+                      intelligentAgent,
+                      'apiDocumentationUrlLabel',
+                    )}
+                  />
+                </Link>
+              ),
+            }}
+          />
+          {filter(
+            currentPlatformFields,
+            field =>
+              get(field, 'label') !== currentPlatformEnablingField,
+            [],
```

Comment:
this would read much clearer if you calculated this above and set it to a variable

Reply:
Done